_overviews/core/futures.md

```diff
@@ -1121,11 +1122,88 @@ behind this is to prevent propagation of critical and control-flow related
+and is also not intended to be handled by user code. By contrast, application code may
+attempt recovery from a "failed" `Future`, which has completed but with an exception.
+
+An execution context can be customized with a reporter that handles fatal exceptions.
+See the factory methods [`fromExecutor`](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext$.html#fromExecutor(e:java.util.concurrent.Executor,reporter:Throwable=%3EUnit):scala.concurrent.ExecutionContextExecutor)
+and [`fromExecutorService`](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext$.html#fromExecutorService(e:java.util.concurrent.ExecutorService,reporter:Throwable=%3EUnit):scala.concurrent.ExecutionContextExecutorService).
+
+Since it is necessary to set the [`UncaughtExceptionHandler`](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Thread.UncaughtExceptionHandler.html)
+for executing threads, as a convenience, `fromExecutor` will create a correctly configured context when passed a `null` executor.
+
+The following example demonstrates how to obtain an `ExecutionContext` with custom error handling
+and also shows the result of different exceptions, as described above:
+
+{% tabs exceptions %}
+{% tab 'Scala 2 and 3' for=exceptions %}
+    import scala.concurrent._, duration._
+    import scala.util.chaining._
+    import java.util.concurrent.ForkJoinPool
+    import java.util.concurrent.TimeUnit.SECONDS
+
+    class C {
+      def crashing(x: Any): Int = throw new NoSuchMethodError("test")
+      def failing(x: Any): Int = throw new NumberFormatException("test")
+      def interrupt(x: Any): Int = throw new InterruptedException("test")
+      def erroring(x: Any): Int = throw new AssertionError("test")
+      // warning: return statement uses an exception to pass control to the caller of the enclosing named method nonlocally
+      def nonlocally(xs: List[Int]): Int = { xs.foreach(x => if (x > 0) return x); -1 }
+      def testCrashes()(implicit ec: ExecutionContext): Future[Int] = Future.unit.map(crashing)
+      def testFails()(implicit ec: ExecutionContext): Future[Int] = Future.unit.map(failing)
+      def testInterrupted()(implicit ec: ExecutionContext): Future[Int] = Future.unit.map(interrupt)
+      def testError()(implicit ec: ExecutionContext): Future[Int] = Future.unit.map(erroring)
+      def testNonLocal()(implicit ec: ExecutionContext): Future[Int] = Future(List(42)).map(nonlocally)
+    }
+
+    object Test extends App {
+      val c = new C
+      def check(f: => Future[Int]): Future[Int] =
+        try Await.ready(f, Duration(1L, SECONDS)).tap(res => println(s"ready $res"))
+        catch { case e: Exception => println(s"threw $e"); Future.never }
+      def reporter(t: Throwable) = println(s"reported $t")
+      locally {
+        import ExecutionContext.Implicits._
+        check(c.testFails())       // a failed Future
```

Comment:
I see that in some tests below what you wrote in the comment corresponds to what was written to the standard output. It would be good to do the same here, I think.

```suggestion
        check(c.testFails())       // ready Failure(NumberFormatException(test))
```